### DISCUSSION TITLE
oAuth2 flow with directus api in an nuxt3 application.

### DISCUSSION THREAD

**Discussion by vanling** [directus/directus#13618] [Archive]
I need someone to rubberduck and brainfart with about oAuth.
I've successfully integrated our oAuth2 provider into the Directus interface and that works perfectly <3 
The next step is integrating the same oAuth provider in the frontend.. via directus 🤔 ?

Everyone on the frontend needs to login via the third-party authorization and either gets an existing linked user from Directus or a new user in Directus

**_So my current theory is_**
- In the Nuxt3 application(frontend) we press login
- We call `directus.url/auth/login/provider-name/` **or do we call the external provider directly?**
- We get redirected to external/third-party login screen
- After a successful login at the third-party we get redirected to the application (server api in nuxt) `/callback?code=..&state=..&action=..` **or are we redirecting to directus api callback first?**
- In the callback code i use the returned code from third party  to complete authentication with directus api?
- i somehow get a token back and use this for api calls etc?

I get a bit stuck in my own head sometimes when it comes to oAuth and callbacks. 
The flow from frontend to the third-party authorization i get and works, I just want to switch to directus authorization...

If someone got a oAuth flow working in their frontend application with Directus as backend, I would love to have your idea's/flow/input

**Comment by aidenfoxx** [comment]
I believe this is covered here: https://docs.directus.io/configuration/sso/#seamless-sso

However, both Directus and your frontend need to be on the same URL (wildcard is acceptable).

**Reply by vanling** [reply]
Thanks! I'm blind, did not read that part as a/the solution. I had just discovered the `?redirect` part by looking in the source and was wondering why that existed.
